js/data.js

```diff
@@ -150,39 +152,33 @@ class Data extends AdaptCollection {
     });
     this.trigger('reset');
     this.trigger('loaded');
-    await Adapt.wait.queue();
+    await wait.queue();
   }
 
   async triggerDataLoaded() {
-    Adapt.log.debug('Firing app:dataLoaded');
+    logging.debug('Firing app:dataLoaded');
     try {
       // Setup the newly added models
       this.forEach(model => model.setupModel?.());
       Adapt.trigger('app:dataLoaded');
     } catch (e) {
-      Adapt.log.error('Error during app:dataLoading trigger', e);
+      logging.error('Error during app:dataLoading trigger', e);
     }
-    await Adapt.wait.queue();
+    await wait.queue();
   }
 
   async triggerDataReady(newLanguage) {
     if (newLanguage) {
       Adapt.trigger('app:languageChanged', newLanguage);
-      this.performStartController();
+      await wait.queue();
     }
-    Adapt.log.debug('Firing app:dataReady');
+    logging.debug('Firing app:dataReady');
     try {
       Adapt.trigger('app:dataReady');
     } catch (e) {
-      Adapt.log.error('Error during app:dataReady trigger', e);
+      logging.error('Error during app:dataReady trigger', e);
     }
-    await Adapt.wait.queue();
-  }
-
-  performStartController() {
```

Comment:
Is this function no longer needed? Surely this changes the behaviour on dataReady?

Reply:
Never mind, I see the new behaviour in the startController now

Reply:
Yea, I just moved the responsibility to where it should be. Sorry for not explaining.